weaviate/collection/classes/grpc.py

```diff
@@ -0,0 +1,140 @@
+
+from google.protobuf import struct_pb2
+from pydantic import BaseModel
+
+from weaviate.collection.classes.internal import _MetadataReturn
+from weaviate.util import BaseEnum
+from weaviate.weaviate_types import UUID
+from weaviate_grpc import weaviate_pb2
+
+
+class HybridFusion(str, BaseEnum):
+    RANKED = "FUSION_TYPE_RANKED"
+    RELATIVE_SCORE = "FUSION_TYPE_RELATIVE_SCORE"
+
+
+@dataclass
+class HybridOptions:
+    alpha: Optional[float] = None
+    vector: Optional[List[float]] = None
+    properties: Optional[List[str]] = None
+    fusion_type: Optional[HybridFusion] = None
+    limit: Optional[int] = None
+    autocut: Optional[int] = None
+
+
+@dataclass
+class GetOptions:
+    limit: Optional[int] = None
+    offset: Optional[int] = None
+    after: Optional[UUID] = None
+
+
+@dataclass
+class BM25Options:
+    properties: Optional[List[str]] = None
+    limit: Optional[int] = None
+    autocut: Optional[int] = None
+
+
+@dataclass
+class NearVectorOptions:
+    certainty: Optional[float] = None
+    distance: Optional[float] = None
+    autocut: Optional[int] = None
+
+
+@dataclass
+class NearObjectOptions:
+    certainty: Optional[float] = None
+    distance: Optional[float] = None
+    autocut: Optional[int] = None
+
+
+@dataclass
+class MetadataQuery:
+    uuid: bool = False
+    vector: bool = False
+    creation_time_unix: bool = False
+    last_update_time_unix: bool = False
+    distance: bool = False
+    certainty: bool = False
+    score: bool = False
+    explain_score: bool = False
+
+
+class LinkTo(BaseModel):
+    link_on: str
+    properties: "PROPERTIES"
+    metadata: MetadataQuery
+
+    def __hash__(self) -> int:  # for set
+        return hash(str(self))
+
+
+class LinkToMultiTarget(LinkTo):
+    target_collection: str
+
+
+PROPERTIES = Union[List[Union[str, LinkTo]], str]
+
+# Can be found in the google.protobuf.internal.well_known_types.pyi stub file but is defined explicitly here for clarity.
+_StructValue: TypeAlias = Union[
+    struct_pb2.Struct,
+    struct_pb2.ListValue,
+    str,
+    float,
+    bool,
+    None,
+    List[float],
+    List[int],
+    List[str],
+    List[bool],
+    List[UUID],
+]
+_PyValue: TypeAlias = Union[
+    Dict[str, "_PyValue"],
+    List["_PyValue"],
+    str,
+    float,
+    bool,
+    None,
+    List[float],
+    List[int],
+    List[str],
+    List[bool],
+    List[UUID],
+]
+_RawObject = Dict[str, _PyValue]
+
+
+@dataclass
+class GrpcResult:
+    metadata: _MetadataReturn
+    result: Dict[str, Union[_StructValue, List["GrpcResult"]]]
+
+
+@dataclass
+class ReturnValues:
+    metadata: Optional[MetadataQuery] = None
+    properties: Optional[PROPERTIES] = None
+
+
+@dataclass
+class RefProps:
+    meta: MetadataQuery
+    refs: Dict[str, "RefProps"]
+
+
+@dataclass
+class SearchResult:
+    properties: weaviate_pb2.ResultProperties
+    additional_properties: weaviate_pb2.ResultAdditionalProps
+
+
+@dataclass
+class SearchResponse:
```

Comment:
same as above

Reply:
It's been moved here to have centralisation of types, it isn't exported from `weaviate.weaviate_classes`, which is where users should access types relevant to them

Reply:
but if we only use it withing the grpc file, I'd just leave it there?